[FSDP] Remove `self.numel_padded_per_param` (unused)

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81243 [FSDP] Move `_post_backward_called` to `_init_param_attributes`
* #80185 [FSDP] Clean up `_lazy_init()`
* #80087 [Easy][FSDP] Add `zero_grad()` to unit test train loop
* **#80002 [FSDP] Remove `self.numel_padded_per_param` (unused)**
* #80000 [FSDP] Move tensor sharding logic to `FlatParamHandle`
* #79984 [FSDP] Deduplicate `_orig_size` and `_unsharded_size`
* #79652 [FSDP] Introduce `FlatParamHandle`

The list `self.numel_padded_per_param` is constructed but never used. This PR removes it.

Differential Revision: [D37726057](https://our.internmc.facebook.com/intern/diff/D37726057)

Commit:
[FSDP] Remove `self.numel_padded_per_param` (unused)

[ghstack-poisoned]

rohan-varma commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -1445,16 +1442,11 @@ def _shard_parameters(self) -> None:
             if orig_storage.size() > 0:
                 orig_storage.resize_(0)  # type: ignore[attr-defined]
 
-        assert len(self.numel_padded_per_param) == len(
```
Does losing this assert make it more likely we might hit bugs/issues?

awgu replied:
It should be very safe to remove since there is an explicit loop over the handles, where there is one parameter per handle, and one append per handle. I think the assert was legacy from the initialize upstream from Fairscale and may have served a purpose at that time.

Commit:
Update on "[FSDP] Remove `self.numel_padded_per_param` (unused)"


The list `self.numel_padded_per_param` is constructed but never used. This PR removes it.

[ghstack-poisoned]

Commit:
Update on "[FSDP] Remove `self.numel_padded_per_param` (unused)"


The list `self.numel_padded_per_param` is constructed but never used. This PR removes it.

[ghstack-poisoned]

awgu commented:
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37726057).

Commit:
Update on "[FSDP] Remove `self.numel_padded_per_param` (unused)"


The list `self.numel_padded_per_param` is constructed but never used. This PR removes it.

Differential Revision: [D37726057](https://our.internmc.facebook.com/intern/diff/D37726057)

[ghstack-poisoned]

awgu commented:
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37726057).

Commit:
Update on "[FSDP] Remove `self.numel_padded_per_param` (unused)"


The list `self.numel_padded_per_param` is constructed but never used. This PR removes it.

Differential Revision: [D37726057](https://our.internmc.facebook.com/intern/diff/D37726057)

[ghstack-poisoned]

awgu commented:
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37726057).

awgu commented:
@pytorchbot merge

awgu commented:
@pytorchbot merge
